src/sharedentitycommunication.hpp

```diff
@@ -124,6 +125,12 @@ class SharedEntityCommunication
        Broadcast ints from master to slaves
        the idea here is if we know size in advance we can
        cut out a communication step
+
+       XXX: This function requires that the entities in ete_diag and
+            ete_offd be ordered consistently with the global true entity
+            ordering on all CPUs. This is the case with ete_diag, so the
+            requirement is reduced to ete_col_map being sorted in increasing
+            order.
```

Comment:
Isn't this always the case for `HYPRE_ParCSRMatrix`?

Reply:
That's the whole question. It seems to be true in practice, but is it guaranteed?

Reply:
I asked Ruipeng about this, and he said if the matrix is created inside hypre (e.g. parmult or transpose), then col_map should be sorted, but if the matrix is created, say, through mfem wrapper, and the input col_map is not sorted, then hypre won't automatically sort it for us. This means it is not guaranteed.